tests/e2e/e2e_gov_test.go

```diff
@@ -0,0 +1,296 @@
+		}
+
+		s.execDistributionFundCommunityPool(s.chainA, 0, sender, tokenAmount.String(), fees.String())
+
+		// there are still tokens being added to the community pool through block production rewards but they should be less than 500 tokens
+		marginOfErrorForBlockReward := sdk.NewInt64Coin(uatomDenom, 500)
+
+		s.Require().Eventually(
+			func() bool {
+				afterDistPhotonBalance, err := getSpecificBalance(chainAAPIEndpoint, distModuleAddress, tokenAmount.Denom)
+				s.Require().NoErrorf(err, "Error getting balance: %s", afterDistPhotonBalance)
+
+				return afterDistPhotonBalance.Sub(beforeDistUatomBalance.Add(tokenAmount.Add(fees))).IsLT(marginOfErrorForBlockReward)
+			},
+			15*time.Second,
+			5*time.Second,
+		)
+	})
+}
+
+func (s *IntegrationTestSuite) verifyChainHaltedAtUpgradeHeight(c *chain, valIdx, upgradeHeight int) {
+	s.Require().Eventually(
+		func() bool {
+			currentHeight := s.getLatestBlockHeight(c, valIdx)
+
+			return currentHeight == upgradeHeight
+		},
+		30*time.Second,
+		5*time.Second,
+	)
+
+	counter := 0
+	s.Require().Eventually(
+		func() bool {
+			currentHeight := s.getLatestBlockHeight(c, valIdx)
+
+			if currentHeight > upgradeHeight {
+				return false
+			}
+			if currentHeight == upgradeHeight {
+				counter++
+			}
+			return counter >= 2
+		},
+		8*time.Second,
+		2*time.Second,
+	)
+}
+
+func (s *IntegrationTestSuite) verifyChainPassesUpgradeHeight(c *chain, valIdx, upgradeHeight int) {
+	s.Require().Eventually(
+		func() bool {
+			currentHeight := s.getLatestBlockHeight(c, valIdx)
+
+			return currentHeight > upgradeHeight
+		},
+		30*time.Second,
+		5*time.Second,
+	)
+}
+
+func (s *IntegrationTestSuite) submitLegacyGovProposal(chainAAPIEndpoint string, sender string, fees string, proposalTypeSubCmd string, proposalId int, proposalPath string) {
+	s.Run("submit_legacy_gov_proposal", func() {
+		s.execGovSubmitLegacyGovProposal(s.chainA, 0, sender, proposalPath, fees, proposalTypeSubCmd)
+
+		s.Require().Eventually(
+			func() bool {
+				proposal, err := queryGovProposal(chainAAPIEndpoint, proposalId)
+				s.Require().NoError(err)
+				return proposal.GetProposal().Status == govv1beta1.StatusDepositPeriod
+			},
+			15*time.Second,
+			5*time.Second,
+		)
+	})
+}
+
+func (s *IntegrationTestSuite) submitNewGovProposal(chainAAPIEndpoint, sender string, proposalId int, proposalPath string) {
+	s.Run("submit_new_gov_proposal", func() {
+		s.execGovSubmitProposal(s.chainA, 0, sender, proposalPath, fees.String())
+
+		s.Require().Eventually(
+			func() bool {
+				proposal, err := queryGovProposal(chainAAPIEndpoint, proposalId)
+				s.T().Logf("Proposal: %s", proposal.String())
+				s.Require().NoError(err)
+
+				return proposal.GetProposal().Status == govv1beta1.StatusDepositPeriod
+			},
+			15*time.Second,
+			5*time.Second,
+		)
+	})
+}
+
+func (s *IntegrationTestSuite) depositGovProposal(chainAAPIEndpoint, sender string, fees string, proposalId int) {
+	s.Run("deposit_gov_proposal", func() {
```

Comment:
IMHO we can use the subtests for an entire test context, like a test in the gov was succeeded. And only print logs in these helper exec methods. WDYT?